cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server_test.go

```diff
@@ -289,29 +303,128 @@ func TestGetAvailablePackageSummaries(t *testing.T) {
-						Context:    &corev1.Context{Namespace: "default"},
+					name:      "jetstack-1",
+					namespace: "ns1",
+					url:       "https://charts.jetstack.io",
+					index:     "testdata/jetstack-index.yaml",
+				},
+			},
+			request: &corev1.GetAvailablePackageSummariesRequest{Context: &corev1.Context{Namespace: "non-default"}},
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackagesSummaries: []*corev1.AvailablePackageSummary{
+					{
+						DisplayName:      "acs-engine-autoscaler",
+						LatestPkgVersion: "2.1.1",
+						IconUrl:          "https://github.com/kubernetes/kubernetes/blob/master/logo/logo.png",
+						ShortDescription: "Scales worker nodes within agent pools",
+						AvailablePackageRef: &corev1.AvailablePackageReference{
+							Identifier: "bitnami-1/acs-engine-autoscaler",
+							Context:    &corev1.Context{Namespace: "default"},
+							Plugin:     &plugins.Plugin{Name: "fluxv2.packages", Version: "v1alpha1"},
+						},
+					},
+					{
+						DisplayName:      "cert-manager",
+						LatestPkgVersion: "v1.4.0",
+						IconUrl:          "https://raw.githubusercontent.com/jetstack/cert-manager/master/logo/logo.png",
+						ShortDescription: "A Helm chart for cert-manager",
+						AvailablePackageRef: &corev1.AvailablePackageReference{
+							Identifier: "jetstack-1/cert-manager",
+							Context:    &corev1.Context{Namespace: "ns1"},
+							Plugin:     &plugins.Plugin{Name: "fluxv2.packages", Version: "v1alpha1"},
+						},
+					},
+					{
+						DisplayName:      "wordpress",
+						LatestPkgVersion: "0.7.5",
+						IconUrl:          "https://bitnami.com/assets/stacks/wordpress/img/wordpress-stack-220x234.png",
+						ShortDescription: "new description!",
+						AvailablePackageRef: &corev1.AvailablePackageReference{
+							Identifier: "bitnami-1/wordpress",
+							Context:    &corev1.Context{Namespace: "default"},
+							Plugin:     &plugins.Plugin{Name: "fluxv2.packages", Version: "v1alpha1"},
+						},
 					},
 				},
+			},
+		},
+		{
+			testName: "uses a filter based on existing repo",
+			testRepos: []testRepoStruct{
+				{
+					name:      "bitnami-1",
+					namespace: "default",
+					url:       "https://example.repo.com/charts",
+					index:     "testdata/valid-index.yaml",
+				},
 				{
-					DisplayName:      "wordpress",
-					LatestPkgVersion: "0.7.5",
-					IconUrl:          "https://bitnami.com/assets/stacks/wordpress/img/wordpress-stack-220x234.png",
-					AvailablePackageRef: &corev1.AvailablePackageReference{
-						Identifier: "bitnami-1/wordpress",
-						Context:    &corev1.Context{Namespace: "default"},
+					name:      "jetstack-1",
+					namespace: "ns1",
+					url:       "https://charts.jetstack.io",
+					index:     "testdata/jetstack-index.yaml",
+				},
+			},
+			request: &corev1.GetAvailablePackageSummariesRequest{
+				Context: &corev1.Context{Namespace: "blah"},
+				FilterOptions: &corev1.FilterOptions{
+					Repositories: []string{"jetstack-1"},
```

Comment:
Not a question about this branch, but prompted by the name of the "repository" search key being `jetstack-1`: If `HelmRepo` objects are namespaced for flux2, the cache key will need to somehow include the namespace to avoid collisions right? (As in, if `myrepo` was in my namespace `foo`, while someone added a `myrepo` `HelmRepo` in their namespace `bar`, then they'd need to have different keys in the cache - perhaps they do, I've not checked the code).

Reply:
yes, the cache key contains namespace, see redisKeyFor() func